client/components/plans/plan-list/index.jsx

```diff
@@ -25,32 +26,40 @@ module.exports = React.createClass( {
 	},
 
 	render: function() {
-		var plans = this.props.plans,
-			showJetpackPlans = false,
+		var className = '',
+			plans = this.props.plans,
+			hideFreePlan = this.props.hideFreePlan ? true : false,
 			isLoadingSitePlans = ! this.props.isInSignup && ! this.props.sitePlans.hasLoadedFromServer,
+			numberOfPlaceholders = 3,
 			site,
 			plansList,
 			currentPlan;
 
+		if ( this.props.sites ) {
```

Comment:
I think we shouldn't set `site` to any value when in signup.
It's a bit confusing but I had a bug because of it in #2374, here is how it appeared:
- Since `site` is defined here, it makes `this.props.site` defined in `Plan`
- Which tried to give a valid `sitePlans` props to `PlanActions` because of this condition https://github.com/Automattic/wp-calypso/blob/update/add-sem-signup-flow/client/components/plans/plan/index.jsx#L110 but fails because `getProductSlug` returns `null`
- Which makes it hide the free trial option here https://github.com/Automattic/wp-calypso/blob/update/add-sem-signup-flow/client/components/plans/plan-actions/index.jsx#L169

Anyway that was a very precise bug but I've seen others related to the differences between the signup and the normal Plans page.
If we want child components in plans to work without a site we should try to avoid giving it when it's not necessary, it will help us find bugs more easily.